src/dodal/devices/pressure_jump_cell.py

```diff
@@ -0,0 +1,236 @@
+
+
+class ValveState(str, Enum):
+    FAULT = "Fault"
+    OPEN = "Open"
+    OPENING = "Opening"
+    CLOSED = "Closed"
+    CLOSING = "Closing"
+
+
+class FastValveState(str, Enum):
+    FAULT = "Fault"
+    OPEN = "Open"
+    OPEN_ARMED = "Open Armed"
+    CLOSED = "Closed"
+    CLOSED_ARMED = "Closed Armed"
+    NONE5 = "Unused"
+
+
+class LimitSwitchState(str, Enum):
+    OFF = "Off"
+    ON = "On"
+
+
+class AllValvesControl(StandardReadable):
+    """
+    valves 2, 4, 7, 8 are not controlled by the IOC,
+    as they are under manual control.
+    """
+
+    def __init__(self, prefix: str, name: str = "") -> None:
+        with self.add_children_as_readables():
+            self.valve_states: DeviceVector[SignalR[ValveState]] = DeviceVector(
+                {i: epics_signal_r(ValveState, f"{prefix}V{i}:STA") for i in [1, 3]}
+            )
+            self.fast_valve_states: DeviceVector[SignalR[FastValveState]] = (
+                DeviceVector(
+                    {
+                        i: epics_signal_r(FastValveState, f"{prefix}V{i}:STA")
+                        for i in [5, 6]
+                    }
+                )
+            )
+
+        with self.add_children_as_readables(ConfigSignal):
+            self.valves_open: DeviceVector[SignalR[bool]] = DeviceVector(
+                {
+                    i: epics_signal_rw(bool, f"{prefix}V{i}:OPENSEQ")
+                    for i in [1, 3, 5, 6]
+                }
+            )
+
+            self.valve_control: DeviceVector[SignalR[ValveControlRequest]] = (
+                DeviceVector(
+                    {
+                        i: epics_signal_rw(ValveControlRequest, f"{prefix}V{i}:CON")
+                        for i in [1, 3, 5, 6]
+                    }
+                )
+            )
+
+        super().__init__(name)
+
+
+class Pump(StandardReadable):
+    def __init__(self, prefix: str, name: str = "") -> None:
+        with self.add_children_as_readables():
+            self.pump_position = epics_signal_r(float, prefix + "POS")
+            self.pump_forward_limit = epics_signal_r(
+                LimitSwitchState, prefix + "D74IN1"
+            )
+            self.pump_backward_limit = epics_signal_r(
+                LimitSwitchState, prefix + "D74IN0"
+            )
+            self.pump_motor_direction = epics_signal_r(
+                PumpMotorDirectionState, prefix + "MTRDIR"
+            )
+            self.pump_speed_rbv = epics_signal_r(int, prefix + "MSPEED_RBV")
+
+        with self.add_children_as_readables(ConfigSignal):
+            self.pump_mode = epics_signal_rw(PumpState, prefix + "SP:AUTO")
+            self.pump_speed = epics_signal_rw(float, prefix + "MSPEED")
+            self.pump_move_forward = epics_signal_rw(bool, prefix + "M1:FORW")
+            self.pump_move_backward = epics_signal_rw(bool, prefix + "M1:BACKW")
+            self.pump_move_backward = epics_signal_rw(
+                PumpMotorControlRequest, prefix + "M1:CON"
```

Comment:
duplicated field name

Reply:
changed the lattern into `pump_connection`